### ISSUE TITLE
Parallel Gateway Convergence Problem

### ISSUE THREAD

**Issue by AgileEduLabs** [ralphhanna/bpmn-server#91]
Hi @ralphhanna, using the Buy Used Car as an example, if the Clean (or Repair) user task branches off to another task, such as "Buy more cleaning supplies", then the Convergent Parallel Gateway (15,17) continues to be in the waiting state even though all flows leading to it has been completed. And so the Drive task can ever be reached.

I've attached the bpmn below.

![ConvergentGatewayProblem](https://github.com/ralphhanna/bpmn-server/assets/69739767/dc0f3911-7778-497a-a36b-95906f83d22e)
[Buy Used Car With Branching.bpmn.txt](https://github.com/ralphhanna/bpmn-server/files/11643269/Buy.Used.Car.With.Branching.bpmn.txt)

<br>
<br>
<br>
<br>
<br>
 
The problem remains if the divergent Inclusive Gateway has been changed to a Parallel Gateway. Again, even though all flows leading to the convergent gateway has been completed, the convergent gateway keeps on waiting.
![ConvergentGatewayProblem2](https://github.com/ralphhanna/bpmn-server/assets/69739767/4f7cc9d9-aef1-4a95-90b8-6c86996c8bd6)

[Buy Used Car With Branching Parallel Gateway.bpmn.txt](https://github.com/ralphhanna/bpmn-server/files/11643282/Buy.Used.Car.With.Branching.Parallel.Gateway.bpmn.txt)

**Comment by ralphhanna** [comment]
Thank you for reporting this bug, but while it is being fixed in the next release you can change your model slightly to work around it.
![image](https://github.com/ralphhanna/bpmn-server/assets/11893416/1dd2c3ec-46dc-43c7-8a5b-92661d31097d)

Thanks

**Comment by AgileEduLabs** [comment]
Hi @ralphhanna, thanks for looking into this. However, when I add the end event, it still doesn't work.

![flowitheventevent](https://github.com/ralphhanna/bpmn-server/assets/69739767/69456f8a-6a33-4687-bea4-0b2c6dbe79d7)

[Buy Used Car with End event.bpmn.txt](https://github.com/ralphhanna/bpmn-server/files/11646495/Buy.Used.Car.with.End.event.bpmn.txt)

**Comment by ralphhanna** [comment]
please remove the gateway between 'Clean' and 'Buy Cleaning...', no need for a gateway just a flow.

Thanks
________________________________

**Comment by AgileEduLabs** [comment]
Hi @ralphhanna, here is what happens when parallel gateway is removed:

The Convergent Parallel Gateway gets fired immediately after completing Clean, even though we are supposed to wait for Repairs before we can Drive
![without integrator](https://github.com/ralphhanna/bpmn-server/assets/69739767/18fbfcca-6120-428b-972d-59e52c0dbf32)
[Buy Used Car 2 without parallel gateway.bpmn.txt](https://github.com/ralphhanna/bpmn-server/files/11646711/Buy.Used.Car.2.without.parallel.gateway.bpmn.txt)

**Comment by ralphhanna** [comment]
Will fix
Thanks

Sent from my iPhone

On Jun 4, 2023, at 5:31 PM, AgileEduLabs ***@***.***> wrote:

﻿

Hi @ralphhanna<https://github.com/ralphhanna>, here is what happens when parallel gateway is removed:

The Convergent Parallel Gateway gets fired immediately after completing Clean, even though we are supposed to wait for Repairs before we can Drive
[without integrator]<https://user-images.githubusercontent.com/69739767/243209200-18fbfcca-6120-428b-972d-59e52c0dbf32.png>
Buy Used Car 2 without parallel gateway.bpmn.txt<https://github.com/ralphhanna/bpmn-server/files/11646711/Buy.Used.Car.2.without.parallel.gateway.bpmn.txt>

**Comment by ralphhanna** [comment]
Sorry for the delay, already fixed. get latest release please
